src/orientation.cpp

```diff
@@ -137,7 +137,8 @@ vector<vector<string>> orientationProbability(Environment& environment) {
     if ((info_int != 0 && info_sign != proba_sign) ||
         (info_int == 0 && proba_sign == -1))
       error = "1";
-
+    if (I3_list[i] == 0)
```

Comment:
The variable `info_int` (l.130) is simply the int version of `I3_list[i]`, so I suppose we should keep using it instead of raw `I3_list[i]` (otherwise it serves for nothing), which means all `I3_list[i]` values in the interval [0, 1) are considered zero for `info_int`.

Then for what Hervé requests, it's enough to delete the second condition (l.138).

Reply:
I thought so at the beginning, but after doing that, I noticed the results changed for non-zero NI3. By doing the way I did, it did not. Maybe it has something to do with the -0.5 in l.130 ([0, 1) being considered zero)? I don't know.

Reply:
If you can verify that those changed non-zero I3 belong to the interval (0, 1), then yes. Now it's the question of whether still using the variable info_int or change to raw value of I3